src/controllers/controller.bar.js

```diff
@@ -288,29 +324,33 @@ module.exports = DatasetController.extend({
 		var me = this;
 		var chart = me.chart;
 		var scale = me._getValueScale();
+		var indexScale = me._getIndexScale();
 		var isHorizontal = scale.isHorizontal();
-		var datasets = chart.data.datasets;
-		var value = scale._parseValue(datasets[datasetIndex].data[index]);
 		var minBarLength = scale.options.minBarLength;
 		var stacked = scale.options.stacked;
 		var stack = me.getMeta().stack;
-		var start = value.start === undefined ? 0 : value.max >= 0 && value.min >= 0 ? value.min : value.max;
-		var length = value.start === undefined ? value.end : value.max >= 0 && value.min >= 0 ? value.max - value.min : value.min - value.max;
-		var i, imeta, ivalue, base, head, size, stackLength;
+		var value = me._getParsedValue(index, scale);
+		var indexValue = me._getParsedValue(index, indexScale);
+		var custom = me._getParsedCustom(index);
+		var start = 0;
+		var end = value;
+		var length = value;
+		var i, imeta, ivalue, base, head, size;
+
+		if (custom && custom.start !== undefined && custom.end !== undefined) {
+			start = custom.start;
+			end = custom.end;
+			length = custom.end - custom.start;
+		}
 
 		if (stacked || (stacked === undefined && stack !== undefined)) {
 			for (i = 0; i < datasetIndex; ++i) {
 				imeta = chart.getDatasetMeta(i);
 
-				if (imeta.bar &&
-					imeta.stack === stack &&
-					imeta.controller._getValueScaleId() === scale.id &&
-					chart.isDatasetVisible(i)) {
-
-					stackLength = scale._parseValue(datasets[i].data[index]);
-					ivalue = stackLength.start === undefined ? stackLength.end : stackLength.min >= 0 && stackLength.max >= 0 ? stackLength.max : stackLength.min;
+				if (imeta.bar && imeta.stack === stack && chart.isDatasetVisible(i)) {
+					ivalue = imeta.controller._getParsedValueByScaleValue(indexValue, indexScale);
 
-					if ((value.min < 0 && ivalue < 0) || (value.max >= 0 && ivalue > 0)) {
+					if ((start <= 0 && end <= 0 && ivalue < 0) || (start >= 0 && end >= 0 && ivalue > 0)) {
```

Comment:
`start < 0 && end < 0` would be equivalent to the old code. Not sure if it was changed on purpose